src/main/java/ip/Duke.java

```diff
@@ -0,0 +1,287 @@
+    private static final String TODO_COMMAND = "todo";
+    private static final String DEADLINE_COMMAND = "deadline";
+    private static final String EVENT_COMMAND = "event";
+    private static final String DELETE_COMMAND = "delete";
+    // Data structures
+    private static final List<Task> taskList = new ArrayList<>();
+
+    private static void loadFile() {
+        try {
+            // Create Duke.txt if it doesn't exist.
+            File f = new File(STORAGE_FILE_PATH);
+            if (!f.exists()) {
+                boolean fileCreated = f.createNewFile();
+                if (!fileCreated) {
+                    throw new IOException();
+                }
+            } else {
+                // Read from Duke.txt and handle it as input.
+                Scanner sc = new Scanner(f);
+                while (sc.hasNext()) {
+                    loadTask(sc.nextLine());
+                }
+                sc.close();
+            }
+        } catch (IOException e) {
+            prettyPrint("Storage file could not be created!");
+        }
+    }
+
+    private static void loadTask(String taskStr) {
+        try {
+            String[] taskArray = taskStr.split("\\|");
+            String taskTypeStr = taskArray[0];
+            String isDoneStr = taskArray[1];
+            String description = taskArray[2];
+
+
+            boolean isDone = isDoneStr.equals("X");
+
+            Task newTask;
+            switch (taskTypeStr) {
+                case "T":
+                    newTask = new ToDo(description, isDone);
+                    break;
+                case "D":
+                    String by = taskArray[3];
+                    newTask = new Deadline(description, by, isDone);
+                    break;
+                case "E":
+                    String at = taskArray[3];
+                    newTask = new Event(description, at, isDone);
+                    break;
+                default:
+                    throw new DukeException(String.format("Could not load task: %s", description));
+            }
+
+            taskList.add(newTask);
+        } catch (DukeException de){
+            prettyPrint(de.toString());
+        }
+
+    }
+
+    private static void storeToFile() {
+        try {
+            FileWriter fw = new FileWriter(STORAGE_FILE_PATH);
+            fw.write("");
+
+            for (Task t: taskList) {
+                fw.append(t.getStorageString());
+                if (!t.equals(taskList.get(taskList.size() - 1))) {
+                    fw.append("\n");
+                }
+            }
+
+            fw.close();
+        } catch (IOException e) {
+            prettyPrint("Failed to save all data!");
+        }
+
+    }
+
+    private static void createTask(String str, TaskType type){
+        try {
+            str = str.strip();
+            Task newTask;
+            String description = "";
+            String extraInfo = "";
+            switch(type) {
+                case TODO:
```

Comment:
Remember to follow the module coding standards! Easily fixed with the IDE configurations.
```suggestion
            switch (type) {
            case TODO:
```